src/components/Store/Store.js

```diff
@@ -13,11 +13,18 @@ const Store = () => {
   const [displayedGames, setDisplayedGames] = React.useState(games);
   const [filter, setFilter] = React.useState('none');
   const [error, setError] = React.useState(null);
+  const [display, setDisplay] = React.useState('grid');
+  const [query, setQuery] = React.useState('');
+  const [wishList, setWishList] = React.useState([]);
 
-  const handleWishList = (gameName, action) => {
-    for (let game of games) {
-      if (game.name === gameName) {
-        action === 'add' ? (game.wishList = true) : (game.wishList = false);
+  const handleWishList = (clickedGame) => {
+    setWishList((prevWishList) => [clickedGame, ...prevWishList]);
```

Comment:
This works, but it's prone to error, it should be done with an if-else. Currently even if the element is to be removed you first add it again at line 21 and after that you remove both ocurences in the loop at line 24.